huggingface_pipelines/speech.py

```diff
@@ -0,0 +1,236 @@
+import os
+import torch
+from typing import Dict, Any
+from dataclasses import dataclass, replace
+from sonar.inference_pipelines.speech import SpeechToEmbeddingModelPipeline
+import tempfile
+import soundfile as sf
+import logging
+from .pipeline import Pipeline, PipelineConfig, PipelineOverwrites
+from .dataset import DatasetConfig
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class HFSpeechToEmbeddingOverwrites(PipelineOverwrites, total=False):
+    """
+    Overwrite options for HFSpeechToEmbeddingPipeline configuration.
+
+    Attributes:
+        encoder_model (str): The name or path of the encoder model to use.
+        fbank_dtype (torch.dtype): The dtype for the fbank features.
+        n_parallel (int): Number of parallel processes for audio processing.
+        pad_idx (int): The index used for padding.
+        audio_column (str): The name of the column containing audio data.
+
+
+    Example :
+
+        overwrites = HFSpeechToEmbeddingOverwrites(
+
+                        encoder = "sonar_basic_text_encoder",
+                        fbank_dtype = torch.float32,
+                        n_parallel = 4,
+                        pad_idx = 2,
+                        audio_column = "audio"
+
+                    )
+
+    """
+    encoder_model: str
+    fbank_dtype: torch.dtype
+    n_parallel: int
+    pad_idx: int
+    audio_column: str
+
+
+@dataclass
+class AudioDatasetConfig(DatasetConfig):
```

Comment:
Where is this class used ?

Reply:
When we initialize audio huggingface datasets